comp/core/status/statusimpl/status.go

```diff
@@ -290,7 +291,10 @@ func (s *statusImplementation) GetStatusBySection(section string, format string,
 			return []byte{}, nil
 		}
 	default:
-		providers := s.sortedProvidersBySection[section]
+		providers, ok := s.sortedProvidersBySection[section]
+		if !ok {
+			return nil, fmt.Errorf("unknown status section '%s'", section)
+		}
```

Comment:
Should we move this check to the `newStatus` constructor and filter all nil providers?

Reply:
`section` is taken as an argument and could be anything, so we need this check here.

But we should also filter nil value of all the FX group we use. I think we could do this in another PR.

Reply:
> section is taken as an argument and could be anything, so we need this check here.

Make sense 😄 

>  I think we could do this in another PR.

Since we are currently doing that for the flare component. I think is best to do it now, since we are currently fixing a common pattern among components that fx groups